rtos/Mutex.cpp

```diff
@@ -55,6 +55,9 @@ osStatus Mutex::lock(uint32_t millisec) {
     if (osOK == status) {
         _count++;
     }
+
+    MBED_ASSERT(status == osErrorTimeout || status == osOK);
```

Comment:
Unfortunately CMSIS-RTOS can also return "osErrorResource" if the mutex is locked and timeout is 0.

And there should be no error if the timeout specified was 0xffffffff. I think.

The maximally-tight assertion is I think:

    MBED_ASSERT(status == osOK ||
                (status == osErrorResource && millisec == 0) ||
                (status == osErrorTimeout && millisec != osWaitForever));
 
Note that if adding an assert here, `trylock_for` can have its removed.